docs/user/gcp/install_upi.md

```diff
@@ -0,0 +1,594 @@
+
+```console
+$ openshift-install create install-config
+? SSH Public Key /home/user_id/.ssh/id_rsa.pub
+? Platform gcp
+? Project ID example-project
+? Region us-east1
+? Base Domain example.com
+? Cluster Name openshift
+? Pull Secret [? for help]
+```
+
+Edit the resulting `openshift-install.yaml` to set `replicas` to 0 for the `compute` pool:
+
+```console
+$ sed -i '1,/replicas: / s/replicas: .*/replicas: 0/' install-config.yaml
+```
+
+Create manifests to get access to the control-plane Machines and compute MachineSets:
+
+```console
+$ openshift-install create manifests
+INFO Consuming "Install Config" from target directory
+```
+
+From the manifest assets, remove the control-plane Machines and the compute MachineSets:
+
+```console
+$ rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml
+$ rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+```
+
+You are free to leave the compute MachineSets in if you want to create compute
+machines via the machine API, but if you do you may need to update the various
+references (`subnetwork`, etc.) to match your environment.
+
+Now we can create the bootstrap Ignition configs.
+
+```console
+$ openshift-install create ignition-configs
+```
+
+After running the command, several files will be available in the directory.
+
+```console
+$ tree
+.
+├── auth
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+├── metadata.json
+└── worker.ign
+```
+
+### Extract infrastructure name from ignition metadata
+
+By default, Ignition generates a unique cluster identifier comprised of the
+cluster name specified during the invocation of the installer and a short
+string known internally as the infrastructure name. These values are seeded
+in the initial manifests within the Ignition configuration. To use the output
+of the default, generated `ignition-configs` extracting the internal
+infrastructure name is necessary.
+
+An example of a way to get this is below:
+
+```console
+$ jq -r .infraID metadata.json
+openshift-vw9j6
+```
+
+Export variables to be used in examples below.
+
+```sh
+export BASE_DOMAIN='example.com'
+export BASE_DOMAIN_ZONE_NAME='example'
+export NETWORK_CIDR='10.0.0.0/16'
+export MASTER_SUBNET_CIDR='10.0.0.0/19'
+export WORKER_SUBNET_CIDR='10.0.32.0/19'
+
+export CLUSTER_NAME=`jq -r .clusterName metadata.json`
+export INFRA_ID=`jq -r .infraID metadata.json`
+export PROJECT_NAME=`jq -r .gcp.projectID metadata.json`
+export REGION=`jq -r .gcp.region metadata.json`
+```
+## Create the VPC
+
+Copy [`01_vpc.py`](../../../upi/gcp/01_vpc.py) locally.
+
+Create a resource definition file: `01_vpc.yaml`
+
+```console
+$ cat <<EOF >01_vpc.yaml
+imports:
+- path: 01_vpc.py
+
+resources:
+- name: cluster-vpc
+  type: 01_vpc.py
+  properties:
+    infra_id: '${INFRA_ID}'
+    region: '${REGION}'
+
+    master_subnet_cidr: '${MASTER_SUBNET_CIDR}'
+    worker_subnet_cidr: '${WORKER_SUBNET_CIDR}'
+EOF
+```
+- `infra_id`: the infra_id from above
```

Comment:
nit: "the infra_id from above" -> "the infrastructure name (`INFRA_ID` above)" or some such, now that you're no longer using `infra_id` in the Markdown text.

Reply:
Done.